precalculator/read.py

```diff
@@ -36,3 +42,33 @@ def get_predictions_from_dataframe(model_id: str, prediction_df: pd.DataFrame) -
         )
         for prediction in predictions
     ]
+
+
+def get_metadata(
+    bucket: str, table_name: str, metadata_key: str, s3_uri: str, model_id: str, start_time: int, end_time: int
+) -> Metadata:
+    try:
+        metadata_obj = s3_client.get_object(Bucket=bucket, Key=metadata_key)
+        metadata_json = json.loads(metadata_obj["Body"].read().decode("utf-8"))
+        metadata = Metadata(**metadata_json)
+    except ClientError as e:
+        if e.response["Error"]["Code"] == "NoSuchKey":
+            metadata = Metadata()
+
+    metadata.model_id = model_id
+    metadata.pipeline_latest_start_time = start_time
+    metadata.preds_last_updated = end_time
+    metadata.pipeline_latest_duration = metadata.preds_last_updated - metadata.pipeline_latest_start_time
+    metadata.pipeline_meta_s3_uri = s3_uri
+
+    model_id_search = dynamodb.scan(
+        TableName=table_name,
+        FilterExpression="SK=:model_id",
+        ExpressionAttributeValues={":model_id": {"S": f"MODELID#{model_id}"}},
+        Select="COUNT",
+    )
```

Comment:
This is gonna be quite an expensive operation since we're only using the SK.

We need to think about how to have a trackable process which can serve as a proxy for successful dynamodb writes. I'd instead save metadata somewhere else during the write process, only when we get a successful write response back from dynamo per-batch. Can chat about this more later.